src/cluster.c

```diff
@@ -4169,7 +4169,8 @@ sds clusterGenNodeDescription(clusterNode *node) {
         (long long) node->ping_sent,
         (long long) node->pong_received,
         nodeEpoch,
-        (node->link || node->flags & CLUSTER_NODE_MYSELF) ?
+        ((node->link && node->link->conn && connGetState(node->link->conn) == CONN_STATE_CONNECTED) ||
```

Comment:
I think a link must always have a connection, so the middle check is redundant. If we're trying to be defensive, I would actually prefer to refactor createClusterLink() so it takes in a connection, to formalize that assumption.

Reply:
I've done two things in the next revision:
1. Refactor createClusterLink() to always take in a connection, and assert that connection cannot be NULL.  
2. Refactor freeClusterLink to also always assert link->conn must not be NULL.